Sources/NIO/BaseSocket.swift

```diff
@@ -348,21 +348,25 @@ class BaseSocket: BaseSocketProtocol {
             precondition(descriptor >= 0, "invalid socket")
         #endif
         self.descriptor = descriptor
+#if !os(Windows)
         do {
             try self.ignoreSIGPIPE()
         } catch {
             self.descriptor = -1 // We have to unset the fd here, otherwise we'll crash with "leaking open BaseSocket"
             throw error
         }
+#endif
     }
 
     deinit {
         assert(!self.isOpen, "leak of open BaseSocket")
     }
 
+#if !os(Windows)
     func ignoreSIGPIPE() throws {
         try BaseSocket.ignoreSIGPIPE(descriptor: self.descriptor)
```

Comment:
The better version of this patch is to edit `BaseSocket.ignoreSIGPIPE`, which already contains platforms-specific code.